File: matiasmasso/Mat.NET/DTO/DTOBank.cs
using MatHelperStd;
using Newtonsoft.Json;
using System;

namespace DTO
{
    public class DTOBank : DTOBaseGuid
    {
        public DTOCountry Country { get; set; }
        public string Id { get; set; }
        public string RaoSocial { get; set; }
        public string NomComercial { get; set; }
        public string Swift { get; set; }
        public string Tel { get; set; }
        public string Web { get; set; }
        [JsonIgnore]
        public Byte[] Logo { get; set; }
        public bool SEPAB2B { get; set; }
        public bool Obsoleto { get; set; }

        public DTOBank() : base()
        {
        }

        public DTOBank(Guid oGuid) : base(oGuid)
        {
        }

        public static DTOBank Factory(DTOCountry oCountry = null/* TODO Change to default(_) if this is not a reference type */, string Id = "")
        {
            DTOBank retval = new DTOBank();
            {
                var withBlock = retval;
                withBlock.Country = oCountry;
                withBlock.Id = Id;
            }
            return retval;
        }

        public static string NomComercialORaoSocial(DTOBank oBank)
        {
            string retval = "";
            if (oBank != null)
            {
                if (oBank.NomComercial == "")
                    retval = oBank.RaoSocial;
                else
                    retval = oBank.NomComercial;
            }
            return retval;
        }

        public bool isSepa()
        {
            bool retval = false;
            if (this.Country != null)
            {
                switch (this.Country.ExportCod)
                {
                    case DTOInvoice.ExportCods.nacional:
                    case DTOInvoice.ExportCods.intracomunitari:
                        {
                            retval = true;
                            break;
                        }
                }
            }
            return retval;
        }

        public static bool validateBIC(string src)
        {
            string pattern = "[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}";
            bool retval = TextHelper.RegexMatch(src, pattern);
            return retval;
        }
    }

}

File: matiasmasso/Mat.NET/DT2/DTOBank.vb
Public Class DTOBank
    Inherits DTOBaseGuid

    Property country As DTOCountry
    Property id As String
    Property raoSocial As String
    Property nomComercial As String
    Property swift As String
    Property tel As String
    Property web As String
    <JsonIgnore> Property logo As Image
    Property SEPAB2B As Boolean
    Property obsoleto As Boolean

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(oGuid As Guid)
        MyBase.New(oGuid)
    End Sub

    Shared Function Factory(Optional oCountry As DTOCountry = Nothing, Optional Id As String = "") As DTOBank
        Dim retval As New DTOBank
        With retval
            .Country = oCountry
            .Id = Id
        End With
        Return retval
    End Function

    Shared Function NomComercialORaoSocial(oBank As DTOBank) As String
        Dim retval As String = ""
        If oBank IsNot Nothing Then
            If oBank.NomComercial = "" Then
                retval = oBank.RaoSocial
            Else
                retval = oBank.NomComercial
            End If
        End If
        Return retval
    End Function

    Public Function IsSepa() As Boolean
        Dim retval As Boolean
        If _Country IsNot Nothing Then
            Select Case _Country.ExportCod
                Case DTOInvoice.ExportCods.Nacional, DTOInvoice.ExportCods.Intracomunitari
                    retval = True
            End Select
        End If
        Return retval
    End Function

    Shared Function ValidateBIC(src As String) As Boolean
        Dim pattern As String = "[A-Z]{6,6}[A-Z2-9][A-NP-Z0-9]([A-Z0-9]{3,3}){0,1}"
        Dim retval As Boolean = TextHelper.RegexMatch(src, pattern)
        Return retval
    End Function
End Class